Buildings/Controls/OBC/Utilities/PIDWithAutotuning/PIDWithAutotuningAmigoFOTD.mo

```diff
@@ -0,0 +1,232 @@
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput trigger
+    "Resets the controller output when trigger becomes true"
+    annotation (Placement(transformation(extent={{-20,-20},{20,20}},rotation=90,origin={-60,-120}), iconTransformation(extent={{-20,-20},{20,20}},rotation=90,origin={-60,-120})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y
+    "Connector for actuator output signal"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),iconTransformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller relay(
+    yHig=yHig,
+    yLow=yLow,
+    deaBan=deaBan)
+    "A relay controller"
+    annotation (Placement(transformation(extent={{22,20},{42,40}})));
+  Buildings.Controls.OBC.Utilities.PIDWithInputGains pid(
+     controllerType=controllerType) "A pid controller"
+    annotation (Placement(transformation(extent={{22,-40},{42,-20}})));
+  Buildings.Controls.OBC.CDL.Continuous.Switch swi
+    "Switch between a PID controller and a relay controller"
+    annotation (Placement(transformation(extent={{62,10},{82,-10}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler samk(y_start=k_start)
+    "Recording the control gain"
+    annotation (Placement(transformation(extent={{-40,-10},{-20,-30}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler samTi(y_start=Ti_start) if with_I
+    "Recording the integral time"
+    annotation (Placement(transformation(extent={{-80,-38},{-60,-58}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler samTd(y_start=Td_start) if with_D
+    "Recording the derivative time"
+    annotation (Placement(transformation(extent={{-40,-60},{-20,-80}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.ControlProcessModel
+    controlProcessModel(
+    yHig=yHig - setPoint,
+    yLow=setPoint + yLow,
+    deaBan=deaBan)
+    "A block to approximate the control process"
+    annotation (Placement(transformation(extent={{-20,40},{-40,60}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.Amigo.PI piParameters
+    if not with_D "A block to calculates the parameters of a PI controller"
+    annotation (Placement(transformation(extent={{-60,70},{-80,90}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.Amigo.PID pidParameters
+    if with_D "A block to calculates the parameters of a PID controller"
+    annotation (Placement(transformation(extent={{-60,40},{-80,60}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.ResponseProcess responseProcess(yHig=yHig
+         - setPoint, yLow=setPoint + yLow)
+    "A block to process the response from the relay controller"
+    annotation (Placement(transformation(extent={{20,40},{0,60}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.ModelTime modTim
```

Comment:
what if the simulation time starts from negative value, say -5000 seconds?

Reply:
The block calculates the time difference and is not sensitive to the absolute value of the time